store/src/commonMain/kotlin/com/dropbox/external/store5/Store.kt

```diff
@@ -0,0 +1,22 @@
+package com.dropbox.external.store5
+
+import kotlinx.coroutines.flow.Flow
+
+typealias Read<Key, Output> = suspend (key: Key) -> Flow<Output?>
+typealias Write<Key, Input> = suspend (key: Key, input: Input) -> Boolean
+typealias Delete<Key> = suspend (key: Key) -> Boolean
+typealias DeleteAll = suspend () -> Boolean
+
+/**
+ * Interacts with a data source.
+ * We recommend [Store] bind to one [Persister]. However, [Store] can bind any source(s) of data.
+ * A [Market] implementation requires at least one [Store]. But typical applications have at least two: one bound to a memory cache and another bound to a database.
+ * @see [Persister].
+ * @see [Market].
+ */
+data class Store<Key, Input, Output>(
```

Comment:
I keep getting hung up on these data classes. It's a bit tough to grok from an api perspective. Most library define some interfaces that show functionality. I'm not sure if uses are used to initing a data class and then passing interfaces into it. I can see this being a function/factory rather than a data class. Same I would expect that function/factory to return an interface and not a real implementation. I think the real interfaces for things like Store are going to make it difficult to test as you can't have a fake store.

Instead I think Store needs an interface that defines all the things it can do

read/write/delete/deletAll. then when you make a RealStore you delegate to a reader/writer/deleter.  This way someone can look at the Store interface and know what functionality a store has.

Reply:
Refactored here https://github.com/MobileNativeFoundation/Store/pull/450/commits/5af29b26a7c60bf21d1dd399b2195f15a271cff1

Reply:
Also refactored:
- https://github.com/MobileNativeFoundation/Store/pull/450/commits/71070330fd038f539f55aeebfd7e983b97aedc7b
- https://github.com/MobileNativeFoundation/Store/pull/450/commits/15dbe22e683bbbe5b9614367c518b29bfdd634b9
- https://github.com/MobileNativeFoundation/Store/pull/450/commits/5f4df24823dd7e1c709921e8836d0a48d6325903
- https://github.com/MobileNativeFoundation/Store/pull/450/commits/aca32636443fc270d5cb8fbd110bc69ebf41b5df